---
title: "Final Project"
author: "Varsha"
date: "11/19/2018"
output:
  pdf_document: default
  html_document: default
---

#Install and Load Libraries required for Featurecounts

```{r}
#BiocManager::install("Rsubread")
#BiocManager::install("DESeq2")
#BiocManager::install("Biobase")
#BiocManager::install("limma")
#BiocManager::install("EnhancedVolcano")
```

```{r}
library(BiocManager)
library(Rsubread)
library(DESeq2)
library(RColorBrewer)
library(gplots)
library(ggplot2)
library(EnhancedVolcano)
library(grid)
library(gridExtra)
library(genefilter)
```

#Set working Directory
```{r}
setwd("/Users/varshaneelakantan/Desktop/test")
```

#RunFeatureCounts
```{r}
#featureCounts(files=c("wtAligned.out.bam", "044_b_sADAligned.out.bam", "044_sADAligned.out.bam" , "131_sADAligned.out.bam" , "131_b_sADAligned.out.bam"),annot.ext="genes.gtf", isGTFAnnotationFile=TRUE,GTF.featureType="exon",GTF.attrType="gene_id")
#fc <- featureCounts(files=c("wtAligned.out.bam", "044_b_sADAligned.out.bam", "044_sADAligned.out.bam" , "131_sADAligned.out.bam" , "131_b_sADAligned.out.bam"),annot.ext="genes.gtf", isGTFAnnotationFile=TRUE,GTF.featureType="exon",GTF.attrType="gene_id")
```


#Save data as a txt file
```{r}
#write.table(x=data.frame(fc$annotation[,c("GeneID","Length")],fc$counts,stringsAsFactors=FALSE),file="readcounts.txt",quote=FALSE,sep="\t",row.names=FALSE)
```


#Differential Analysis using DESeq
##Reading Data
```{r}
dds <- read.delim("featurecounts_output.txt" , sep = "\t" , header=TRUE, row.names = 1)  #Removed Duplicates from the file or else error keeps popping up
dds <- dds[-c(1)] #Ignoring the Length column
data <- as.data.frame(dds)
coldata <- read.delim("coldata.txt", sep = "\t") #Made an excel with the column names of the different samples
coldata <- as.data.frame(coldata)
View(coldata)
```


##Run DESEQ
```{r}
dds_output <- DESeqDataSetFromMatrix(countData = data,
                              colData = coldata,
                              design = ~ Condition)

deseq_dds <- DESeq(dds_output)
res_test2 <- results(deseq_dds, contrast = c("Condition", "Control" , "test2"))
res_test1 <- results(deseq_dds, contrast = c("Condition", "Control" , "test1"))

resa <- results(deseq_dds, contrast = c("Condition", "test1" , "Control"))
resb <- results(deseq_dds, contrast = c("Condition", "test2" , "Control"))
res_sad1_sad2 <- results(deseq_dds, contrast = c("Condition", "test2" , "test1"))
```
#Summary of the results
```{r}
#mcols(res_test1, use.names=TRUE)
#mcols(res_test2, use.names=TRUE)
#sizeFactors(deseq_dds) #The size factors sj are accessible via sizeFactors
```

#ReOrdering based on Lowest p value
```{r}
resOrdered_1 <- res_test1[order(res_test1$pvalue),]
resOrdered_2 <- res_test2[order(res_test2$pvalue),]
resOrdered_sAD1vsctrl <- resa[order(resa$pvalue),]
resOrdered_sAD2vsctrl <- resb[order(resb$pvalue),]
resOrdered_sAD2vssAD1 <- res_sad1_sad2[order(res_sad1_sad2 $pvalue),]
```
#Setting a cutoff p value 
```{r}
#sum(resOrdered_1$padj < 0.1, na.rm=TRUE)
#sum(resOrdered_2$padj < 0.1, na.rm=TRUE)
#res05 <- results(deseq_dds, alpha=0.05)
#summary(res05)
#resultsNames(deseq_dds)
```

#Exporting data into csv files
```{r}
#write.csv(as.data.frame(resOrdered_1), file="dds_ctrlvssAD1.csv")
#write.csv(as.data.frame(resOrdered_2), file="dds_ctrlvssAD2.csv")
#write.csv(as.data.frame(resOrdered_sAD1vsctrl), file="dds_sAD1vsctrl.csv")
#write.csv(as.data.frame(resOrdered_sAD2vsctrl), file="dds_sAD2vsctrl.csv")
#write.csv(as.data.frame(resOrdered_sAD2vssAD1), file="dds_sAD2vssAD1.csv")

```

#Generating plots fpr data distributions
##Plot for how the data is dispersed
```{r}
plotDispEsts(deseq_dds, ylim = c(1e-6, 1e1) )
```

#To make it rlog transformed data

```{r}
rld <- rlog(deseq_dds)
head( assay(rld) )
```
#Generating PLots from R transformed data
##Heatmap of Euclidean sample distances after rlog transformation
###Data used is rlog transformed
```{r}
sampleDists <- dist( t( assay(rld) ) )
sampleDists
sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <- paste( rld$treatment,
                                     rld$patient, sep="-" )
colnames(sampleDistMatrix) <- NULL

colours = colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
heatmap.2( sampleDistMatrix, trace="none", col=colours)
```


##Heatmap for top 50 variable genes
###Data used is rlog transformed
```{r}
topVarGenes <- head( order( rowVars( assay(rld) ), decreasing=TRUE ), 50 )
heatmap.2( assay(rld)[ topVarGenes, ], scale="row",
           trace="none", dendrogram="column",
           col = colorRampPalette( rev(brewer.pal(9,"RdBu")) )(255))
```

##Volcano plot (Using Enhanced volcano plots package by Kevin Blighe)
###Data used is the original results from deseq analysis

```{r}
p1 <- EnhancedVolcano(resa,
                      lab = rownames(resa),
                      x = "log2FoldChange",
                      y = "pvalue",
                      pCutoff = 10e-8,
                      FCcutoff = 2.0,
                      xlim = c(-10,10),
                      ylim = c(0, 200),
                      transcriptLabSize = 2.5,
                      title = "sADa Versus Control",
                      colAlpha = 1,
                      legendPosition = "bottom",
                      legendLabSize = 10,
                      legendIconSize = 3.0,
                      border = "full",
                      borderWidth = 1,
                      borderColour = "black",
                      gridlines.major = FALSE,
                      gridlines.minor = FALSE)
grid.arrange(p1, ncol=2)
grid.rect(gp=gpar(fill=NA))

p2 <- EnhancedVolcano(resb,
                      lab = rownames(resb),
                      x = "log2FoldChange",
                      y = "pvalue",
                      xlab = bquote(~Log[2]~ "fold change"),
                      ylab = bquote(~-Log[10]~italic(P)),
                      pCutoff = 10e-8,
                      FCcutoff = 2.0,
                      xlim = c(-10,10),
                      ylim = c(0, 300),
                      transcriptLabSize = 2.5,
                      title = "sADb Versus Control",
                      colAlpha = 1,
                      legend=c("NS","Log2 FC","P","P & Log2 FC"),
                      legendPosition = "bottom",
                      legendLabSize = 10,
                      legendIconSize = 3.0, 
                      border = "full",
                      borderWidth = 1,
                      borderColour = "black",
                      gridlines.major = FALSE,
                      gridlines.minor = FALSE)
grid.arrange(p2, ncol=2)
grid.rect(gp=gpar(fill=NA))
p3 <- EnhancedVolcano(res_sad1_sad2,
                      lab = rownames(res_sad1_sad2),
                      x = "log2FoldChange",
                      y = "pvalue",
                      xlab = bquote(~Log[2]~ "fold change"),
                      ylab = bquote(~-Log[10]~italic(P)),
                      pCutoff = 10e-8,
                      FCcutoff = 2.0,
                      xlim = c(-10,10),
                      ylim = c(0, 300),
                      transcriptLabSize = 2.5,
                      title = "sADb Versus sADa",
                      colAlpha = 1,
                      legend=c("NS","Log2 FC","P","P & Log2 FC"),
                      legendPosition = "bottom",
                      legendLabSize = 10,
                      legendIconSize = 3.0, 
                      border = "full",
                      borderWidth = 1,
                      borderColour = "black",
                      gridlines.major = FALSE,
                      gridlines.minor = FALSE)
grid.arrange( p3, ncol=2)
grid.rect(gp=gpar(fill=NA))

```


